apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/profile/ProfileTaskExecutionContext.java

```diff
@@ -48,7 +48,7 @@ public class ProfileTaskExecutionContext {
 
     public ProfileTaskExecutionContext(ProfileTask task) {
         this.task = task;
-        profilingSegmentSlots = new AtomicReferenceArray<>(Config.Profile.MAX_PARALLEL);
+        profilingSegmentSlots = new AtomicReferenceArray<>(Config.Profile.MAX_PARALLEL * (Config.Profile.MAX_ACCEPT_SUB_PARALLEL + 1));
```

Comment:
I think this may not current. This would cause `if (profilingEndpointCount >= Config.Profile.MAX_PARALLEL) {` true even if the slots are still available.

Reply:
The `profilingEndpointCount`  is only the count of the profiled endpoint, not including the sub-tasks.
Also, If we need the sub-process(`continue`) profiling, need to use `ProfileTaskExecutionContext#continueProfiling
` instance of `ProfileTaskExecutionContext#attemptProfiling` method.